### PR TITLE
fix(options): Stryker silently runs with default config if custom config file is set but not found

### PR DESCRIPTION
**Pull request by ggingras**
To fix this issue, I thought that an InputException should only be thrown if the invalid config was provided by the user.

If no config is provided and the default config is not present, do as before and silently continue.

Note: This is my first PR for the Stryker project, do not hesitate to suggest changes.

### CODE REVIEW

**Comment by rouke-broersma** [comment]
Hi @ggingras 

Could you explain the problem? If no custom config is provided stryker will already use defaults. Am I misunderstanding the scenario?

**Comment by ggingras** [comment]
> Hi @ggingras
> 
> Could you explain the problem? If no custom config is provided stryker will already use defaults. Am I misunderstanding the scenario?

It is to address this issue: https://github.com/stryker-mutator/stryker-net/issues/2008 log by another user, but I also faced the same issue while working with Stryker. In my use case, it was a relative path issue.

File: src/Stryker.CLI/Stryker.CLI/ConfigReader.cs [hunk 1]
```diff
@@ -24,12 +25,20 @@ public void Build(IStrykerInputs inputs, string[] args, CommandLineApplication a
...
-            var configFilePath = Path.Combine(basePath, cmdConfigHandler.GetConfigFilePath(args, app));
+            var configFilePath = Path.Combine(basePath, configFileOption?.Value() ?? "stryker-config.json");
             if (File.Exists(configFilePath))
             {
                 JsonConfigHandler.DeserializeConfig(configFilePath, inputs);
             }
+            else if (configFileOption.HasValue())
+            {
+                // only throw exception if the config file path provided by the user
+                throw new InputException($"Config file not found {configFilePath}");
```
Comment (psfinaki) [inline src/Stryker.CLI/Stryker.CLI/ConfigReader.cs:39]:
```suggestion
                throw new InputException($"Config file not found at {configFilePath}");
```

File: src/Stryker.CLI/Stryker.CLI/ConfigReader.cs [hunk 2]
```diff
@@ -24,12 +25,20 @@ public void Build(IStrykerInputs inputs, string[] args, CommandLineApplication a
...
             // read config from json and commandline
-            var configFilePath = Path.Combine(basePath, cmdConfigHandler.GetConfigFilePath(args, app));
+            var configFilePath = Path.Combine(basePath, configFileOption?.Value() ?? "stryker-config.json");
             if (File.Exists(configFilePath))
             {
                 JsonConfigHandler.DeserializeConfig(configFilePath, inputs);
             }
+            else if (configFileOption.HasValue())
+            {
+                // only throw exception if the config file path provided by the user
```
Comment (psfinaki) [inline src/Stryker.CLI/Stryker.CLI/ConfigReader.cs:38]:
```suggestion
                // only throw exception if the config file path is provided by the user
```

**Review by psfinaki — DISMISSED** [review]
Thanks a lot!

File: src/Stryker.CLI/Stryker.CLI/ConfigReader.cs [hunk 3]
```diff
@@ -24,12 +25,20 @@ public void Build(IStrykerInputs inputs, string[] args, CommandLineApplication a
...
             // read config from json and commandline
-            var configFilePath = Path.Combine(basePath, cmdConfigHandler.GetConfigFilePath(args, app));
+            var configFilePath = Path.Combine(basePath, configFileOption?.Value() ?? "stryker-config.json");
             if (File.Exists(configFilePath))
             {
                 JsonConfigHandler.DeserializeConfig(configFilePath, inputs);
             }
+            else if (configFileOption.HasValue())
+            {
+                // only throw exception if the config file path is provided by the user
```
Comment (rouke-broersma) [inline src/Stryker.CLI/Stryker.CLI/ConfigReader.cs:38]:
```suggestion
                // throw exception if the config file path is provided by the user yet it does not exist
```

**Pull request merged**